numpy/lib/financial.py

```diff
@@ -22,17 +24,20 @@
                 'start':1,
                 'finish':0}
 
-def _convert_when(when):
+def _convert_when(when, sentinel=None):
```

Comment:
Why are you changing the `when` argument here?

Reply:
I'm changing it to ensure that the called/cached values are always of the correct type. 

Since this function is called in almost all of the financial functions this was the easiest way to ensure all of them used the correct where values. 

The real issue is that if the `where_num`, as I've called it, is not a Decimal type, then the calculations are broken later because it won't be `Decimal('0')` or `Decimal('1')`. It'll end up being `1.0`, or `0.0` in the calculations, which then causes a type error because `Decimal` and `float` are not friends.

Reply:
But id doesn't end up as `0.0` or `1.0`, does it? It returns an integer instead.